From 2a2bfef937db6e0bd79a457b5762ddee0a6e6c97 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Tue, 23 Nov 2021 17:58:52 +0100
Subject: [PATCH 1/3] riscv: T-HEAD PLIC in device-tree

The vendor OpenSBI used

    compatible = "allwinner,sun20i-d1-plic"

to identify a T-HEAD PLIC. Upstream OpenSBI uses

    compatible = "thead,c900-plic".

The device-tree should to take this into account:

    plic: interrupt-controller@10000000 {
        compatible = "allwinner,sun20i-d1-plic",
                     "thead,c900-plic",
                     "sifive,plic-1.0.0";

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
---
 arch/riscv/dts/sun20i-d1.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/riscv/dts/sun20i-d1.dtsi b/arch/riscv/dts/sun20i-d1.dtsi
index 280e0336d4..56d02c33b0 100644
--- a/arch/riscv/dts/sun20i-d1.dtsi
+++ b/arch/riscv/dts/sun20i-d1.dtsi
@@ -1136,7 +1136,7 @@
 
 		plic: interrupt-controller@10000000 {
 			compatible = "allwinner,sun20i-d1-plic",
-				     "sifive,plic-1.0.0";
+				     "thead,c900-plic";
 			reg = <0x10000000 0x4000000>;
 			#address-cells = <0>;
 			interrupts-extended = <&cpu0_intc 11>,
-- 
2.32.0

